lib/ddtrace/contrib/redis/patcher.rb

```diff
@@ -59,8 +61,10 @@ def call_pipeline(*args, &block)
               pin.tracer.trace(Datadog::Contrib::Redis::Ext::SPAN_COMMAND) do |span|
                 span.service = pin.service
                 span.span_type = Datadog::Contrib::Redis::Ext::TYPE
-                commands = args[0].commands.map { |c| Datadog::Contrib::Redis::Quantize.format_command_args(c) }
-                span.resource = commands.join("\n")
+                if datadog_configuration[:show_command_args]
+                  commands = args[0].commands.map { |c| Datadog::Contrib::Redis::Quantize.format_command_args(c) }
+                  span.resource = commands.join("\n")
+                end
```

Comment:
I think at the very least we want an `else` case where `span.resource = Datadog::Contrib::Redis::Ext::SPAN_COMMAND`. This fallback occurs at the agent-level in our ingest pipeline anyway afaik, but it's better to explicitly set it rather than having to rely on normalization code.

More broadly though, I think it's worth considering whether at least capturing the operation `get ? / set ? / etc` is preferable here to not capturing anything? This is what we do for `auth` commands currently, https://github.com/DataDog/dd-trace-rb/blob/750ebd9d820d433a80d011b2988bb26599c064a5/lib/ddtrace/contrib/redis/quantize.rb#L24.  If that can be accomplished in a performant way, would that still alleviate any security concerns? The `span.resource` provides a lot of value in the datadog UI, as we generate hits/latency/error statistics per (among other things) unique env/service/resource combination, and so if every resource is the same, the value of those redis service metrics are much lower.

Wdyt?

Reply:
Good point, the first arg is pretty important. I'll play around with some approaches and add the best way in.

Reply:
I think the code path you are looking for is in `Datadog::Contrib::Redis::Quantize`.

You should be able to tweak it to be able to retrieve the command name, but not the arguments.

Just to reinforce what @ericmustin said, if the command is, for example, `ACL SETUSER user1 on allkeys +set`, we'd still like to capture `ACL SETUSER`.

Reply:
Thank you @marcotc & @ericmustin - just pushed some changes. Let me know what you think.